### DISCUSSION TITLE
r.js压缩layui  但是还是会单独出现

### DISCUSSION THREAD

**Discussion by ken678** [layui/layui#1899] [问答]
正常requireJS使用没有问题 现在就是r.js打包压缩有问题
![image](https://github.com/layui/layui/assets/15792472/69ed13dc-a30f-4390-a7e8-c7b749c04600)
require-backend.min.js里面已经含有所有代码包括layui代码  但是还会单独出来
![image](https://github.com/layui/layui/assets/15792472/55234ae5-5156-42c6-b2a8-5ceb2ed87cb5)
进行压缩也会报警告
has more than one anonymous define. May be a built file from another build system like, Ender. Skipping normalization.
![image](https://github.com/layui/layui/assets/15792472/12e1b457-5356-4ab4-8d2f-090192d4e9c2)

@Sight-wcg

**Comment by ken678** [comment]
目前暂时只能先这样了
    excludeShallow: [ // 排除的模块列表
        'layui'
    ]

**Comment by Sight-wcg** [comment]
我没有 AMD 模块相关库或工具的使用经验，爱莫能助了。

**Reply by ken678** [reply]
好的感谢回复
